rust/kernel/chrdev.rs

```diff
@@ -20,10 +19,64 @@ use crate::error::{Error, KernelResult};
 use crate::file_operations;
 use crate::types::CStr;
 
+/// Character device.
+///
+/// # Invariants
+///
+/// - [`self.0`] is valid and non-null.
+/// - [`(*self.0).ops`]   is valid and has static lifetime.
+/// - [`(*self.0).owner`] is valid and has static lifetime.
```

Comment:
I don't think we can say it always has a static lifetime since this one might be null.

```suggestion
/// - [`(*self.0).owner`] is valid and has static lifetime if non-null.
```

Reply:
For the C kernel side, it's valid for `owner` to be null. But with out `Cdev` Rust class, I don't think it can ever be null, because it's passed in via `module: &'static crate::ThisModule`. We might change this in the future, of course.

Reply:
See my other comment regarding this https://github.com/Rust-for-Linux/linux/pull/207#discussion_r617972400. It can be null because `ThisModule.0` is designed to be null when built-in (like `THIS_MODULE` in the C side). The `'static` lifetime does not help here, because that applies only to `ThisModule`, not to its pointer.

Reply:
Yes, noticed my error so I deleted the message, but you had already read it :)